hoomd/pytest/test_device.py

```diff
@@ -138,3 +138,36 @@ def test_device_notice(device, tmp_path):
     if device.communicator.rank == 0:
         with open(device.msg_file) as fh:
             assert fh.read() == ""
+
+
+def test_noticefile(device, tmp_path):
+
+    # Message file declared. Should output in specified file.
+    device.message_filename = str(tmp_path / "str_message")
+    msg = "This message should output."
```

Comment:
These tests are failing with the changes I suggested to `write`. You need to include `'\n'` in the messages to write. Also, it would be helpful if each `assert` as in a separate `test_noticefile_*(device, tmp_path)` function.